helper/tlsutil/config.go

```diff
@@ -146,8 +147,18 @@ func (c *Config) AppendCA(pool *x509.CertPool) error {
 		return fmt.Errorf("Failed to read CA file: %v", err)
 	}
 
+	block, _ := pem.Decode([]byte(data))
```

Comment:
The PEM file can have multiple certificates in it. You are only parsing the first. See: https://golang.org/src/crypto/x509/cert_pool.go?s=2791:2855#L102